components/gfx/display_list/mod.rs

```diff
@@ -484,6 +507,48 @@ impl DisplayList {
                          self.background_and_borders.iter().rev())
 
     }
+
+    /// Returns the PaintLayer in the given DisplayList with a specific layer ID.
+    pub fn find_layer_with_layer_id(&self, layer_id: LayerId) -> Option<Arc<PaintLayer>> {
+        for kid in &self.layered_children {
+            if let Some(paint_layer) = PaintLayer::find_layer_with_layer_id(&kid, layer_id) {
+                return Some(paint_layer);
+            }
+        }
+
+        for kid in &self.children {
+            if let Some(paint_layer) = kid.display_list.find_layer_with_layer_id(layer_id) {
+                return Some(paint_layer);
+            }
+        }
+
+        None
+    }
+
+    pub fn bounds(&self) -> Rect<Au> {
+        let union_all_items = |list: &LinkedList<DisplayItem>, mut bounds: Rect<Au>| -> Rect<Au> {
```

Comment:
Does this need to be a closure? Can it just be `fn union_all_items(list: ..., mut bounds: ...) -> Rect<Au> { ... }`?